src/drive/web/modules/queries.js

```diff
@@ -355,3 +355,17 @@ export const buildFolderByPathQuery = path => ({
     fetchPolicy: defaultFetchPolicy
   }
 })
+
+export const buildNewSharingShortcutQuery = () => ({
+  definition: () =>
+    Q('io.cozy.files')
+      .where({
+        class: 'shortcut',
+        'metadata.sharing.status': 'new'
+      })
+      .indexFields(['class', 'metadata.sharing.status']),
```

Comment:
I think we should invert the both fields. `class` is less restrictive than `metatada.sharing.status` and to have better performance we should always put first in the index the more restrictive field.